### DISCUSSION TITLE
I have a problem with the graphics, it does not detect the corner radius in any widget.

### DISCUSSION THREAD

**Discussion by Rmollocuaquira** [TomSchimansky/CustomTkinter#2845] [Q&A]
I followed the instructions for use but the examples look different from the examples:

![Captura de pantalla de 2024-06-12 19-57-27](https://github.com/TomSchimansky/CustomTkinter/assets/75807563/4fa5421a-cc43-4116-b3ee-b576e8631796)

the rounded corners of buttons, sliders and other widgets look wrong. it's as if the corner_radius parameter doesn't work and everything shows me straight. I tried the package on another device and it works fine but not on this one which is the main one.

I reinstalled the whole Ubuntu 22.04 LTS operating system, but I still have the problem so I am looking for help here.  Thanks in advance.

**Comment by dipeshSam** [comment]
Although, this bug is known and need to be fixed by @TomSchimansky .

But for instant fix, see the following thread: https://github.com/TomSchimansky/CustomTkinter/issues/2286#issue-2158676584

**Comment by Rmollocuaquira** [comment]
> Although, this bug is known and need to be fixed by @TomSchimansky .
> 
> But for instant fix, see the following thread: [#2286 (comment)](https://github.com/TomSchimansky/CustomTkinter/issues/2286#issue-2158676584)

thanks for answering, but although the error you mentioned is similar it is not the same, since in this case I am not using any anaconda environment, only python's venv, a notable difference between that error and mine is that in my case the text fonts load correctly but the rounded borders do not.

**Comment by dipeshSam** [comment]
This might be system specific. Other approach to solve the problem could be trying different drawing methods. `CustomTkinter` uses different drawing methods for different operating systems.

``` Python
preferred_drawing_method: str = None  # 'polygon_shapes', 'font_shapes', 'circle_shapes'
```

Sample Code:
``` Python
from customtkinter import CTk, CTkButton, DrawEngine


if __name__ == "__main__":
    app = CTk()
    app.geometry("400x300")
    
    DrawEngine.preferred_drawing_method="polygon_shapes"
    button1 = CTkButton(app, text="Button using 'polygon_shapes'", width=250, height=50, corner_radius=20)
    button1.place(relx=0.5, anchor="center", rely=0.2)
    
    DrawEngine.preferred_drawing_method="font_shapes"
    button2 = CTkButton(app, text="Button using 'font_shapes'", width=250, height=50, corner_radius=20)
    button2.place(relx=0.5, anchor="center", rely=0.5)
    
    DrawEngine.preferred_drawing_method="circle_shapes"
    button3 = CTkButton(app, text="Button using 'circle_shapes'", width=250, height=50, corner_radius=20)
    button3.place(relx=0.5, anchor="center", rely=0.8)
    

    app.mainloop()
```
Output:
![ctk_prefered_drawing_methods](https://github.com/TomSchimansky/CustomTkinter/assets/72375763/7a49a4ef-7508-472b-8505-7295975d81f3)

In this photograph demonstration, it can be clearly seen, that different drawing methods causes different smoothness.
Try these, might work. Otherwise, wait until @TomSchimansky come back.

Regards

**Comment by Rmollocuaquira** [comment]
I tried the solution but I still have the problem with the graphics, I hope @TomSchimansky can solve it soon.
